File: 프로그래머스/lv2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];

        int i=1;
        while(true){
            int width = Math.max(i, yellow/i);
            int height = Math.min(i, yellow/i);
            if(width * height == yellow){
                if((width+2)*2 + (height*2) == brown){
                    answer[0] = width+2;
                    answer[1] = height+2;
                    break;
                }
                else i++;
            }
            else i++;
        }
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    
    for (let i = 1; i <= yellow; i++) {
        if (yellow % i !== 0) continue;
        
        let j = yellow / i;
        
        if ((i + 2) * (j + 2) === brown + yellow) {
            return [j + 2, i + 2]
        }
    }
}